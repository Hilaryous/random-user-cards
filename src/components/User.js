// @flow
import React, { PureComponent } from 'react'
import styled from 'react-emotion'
import Card from './Card'
import Text from './Text'
import type { UserType } from '../types/User'

const UserCard = styled(Card)({
  display: 'grid',
  gridTemplateColumns: '106px 1fr',
  gridGap: 20,
  paddingTop: 50,
  paddingBottom: 50,
}, ({ border }) => ({
  border,
}))

const UserInfo = styled('div')({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'space-around',
})

type Props = {
  currentUserId: string,
  toggleUser: (string) => void,
} & UserType

class User extends PureComponent<Props> {
  handleToggleUser = () => {
    const { id, toggleUser } = this.props
    toggleUser(id)
  }

  render() {
    const { currentUserId, id, avatar, catchPhrase, email, name } = this.props
    return (
      <UserCard border={currentUserId === id ? '1px solid' : 'none'} onClick={this.handleToggleUser}>
        <img alt="User Avatar" src={avatar} />
        <UserInfo>
          <Text modifier="italic">
            {`\u0022${catchPhrase}\u0022`}
          </Text>
          <div>
            <Text modifier="bold">
              {name}
            </Text>
            <Text>
              {email}
            </Text>
          </div>
        </UserInfo>
      </UserCard>
    )
  }
}

export default User
