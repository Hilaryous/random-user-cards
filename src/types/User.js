// @flow
export type UserType = {
  avatar: string,
  catchPhrase: string,
  email: string,
  id: string,
  name: string,
}
