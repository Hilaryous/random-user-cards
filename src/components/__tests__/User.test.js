/* eslint-disable function-paren-newline */
import React from 'react'
import renderer from 'react-test-renderer'
import User from '../User'

describe('User', () => {
  it('shows the correct snapshot tree for the User component, selected', () => {
    const tree = renderer.create(
      <User
        currentUserId="1"
        toggleUser={() => {}}
        id="1"
        avatar="foo.bar.com"
        catchPhrase="i'm cool!"
        email="baz@bar.com"
        name="Foo Bar"
      />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot tree for the User component, unselected', () => {
    const tree = renderer.create(
      <User
        currentUserId="1"
        toggleUser={() => {}}
        id="3"
        avatar="foo.bar.com"
        catchPhrase="i'm cool!"
        email="baz@bar.com"
        name="Foo Bar"
      />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
