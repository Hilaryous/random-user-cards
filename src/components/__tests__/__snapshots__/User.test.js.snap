// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User shows the correct snapshot tree for the User component, selected 1`] = `
<li
  className="css-gxxquy"
  onClick={[Function]}
>
  <img
    alt="User Avatar"
    src="foo.bar.com"
  />
  <div
    className="css-16p1w4b"
  >
    <p
      className="css-18fexks"
    >
      "i'm cool!"
    </p>
    <div>
      <p
        className="css-11losmg"
      >
        Foo Bar
      </p>
      <p
        className="css-tku0cv"
      >
        baz@bar.com
      </p>
    </div>
  </div>
</li>
`;

exports[`User shows the correct snapshot tree for the User component, unselected 1`] = `
<li
  className="css-15acy4d"
  onClick={[Function]}
>
  <img
    alt="User Avatar"
    src="foo.bar.com"
  />
  <div
    className="css-16p1w4b"
  >
    <p
      className="css-18fexks"
    >
      "i'm cool!"
    </p>
    <div>
      <p
        className="css-11losmg"
      >
        Foo Bar
      </p>
      <p
        className="css-tku0cv"
      >
        baz@bar.com
      </p>
    </div>
  </div>
</li>
`;
